src/checkpoint/checkpoint.ts

```diff
@@ -70,32 +63,86 @@ export default class Checkpoint {
 
   public async start() {
     this.log.debug('starting');
+
+    // TODO(perfectmak): Validate config before starting
+
+    await this.store.createStore();
+
     const blockNum = await this.getStartBlockNum();
     return await this.next(blockNum);
   }
 
   public async reset() {
     this.log.debug('reset');
+
+    await this.store.deleteStore();
```

Comment:
I think we should not deleting the checkpoints here, only the latest sync block. Checkpoints wont change (unless using different network) and will be used to sync faster the second time you replay.

Reply:
Hmmm... True, checkpoints shouldn't change. I asked a similar question in my comment [here](https://github.com/snapshot-labs/sx-api/pull/39/files#r854465971), about what `resets()` use case is meant for. I can update this to not delete the whole store but only unset the `last_indexed_block` metadata. That also means there won't be a need to call `this.store.createStore()` subsequently, as the `start()` method will handle that part.

Let me know if this is okay?

Reply:
Sound good!